subsys/net/buf.c

```diff
@@ -346,8 +346,8 @@ struct net_buf *net_buf_alloc_len(struct net_buf_pool *pool, size_t size,
 	net_buf_reset(buf);
 
 #if defined(CONFIG_NET_BUF_POOL_USAGE)
-	pool->avail_count--;
-	__ASSERT_NO_MSG(pool->avail_count >= 0);
+	atomic_dec(&pool->avail_count);
+	__ASSERT_NO_MSG(atomic_get(&pool->avail_count) >= 0);
```

Comment:
Isn't atomit_t an unsigned type, i.e. the assert above will never be false?

Reply:
Or rather atomic_val_t (which atomic_get() returns)

Reply:
according to atomic.h:
typedef int atomic_t;

Reply:
oh, ok, my mistake then. Then I don't have any objections to this PR